app/controllers/dor_controller.rb

```diff
@@ -47,24 +47,36 @@ def label
     respond_with params.merge('label' => Dor::MetadataService.label_for(params[:source_id]))
   end
 
+  ##
+  # reindex will make a synchronous index update to the Solr core
+  #
+  # @option params [String] `:pid` the druid for the object to reindex
+  # @option params [String] `:asynchronous` if present does an asynchronous transaction with solr index (i.e., no commit)
   def reindex
-    obj = Dor.load_instance params[:pid]
+    obj = Dor.load_instance params[:pid] # could be any type of Dor object
```

Comment:
FWIW:  when i was trying to wrap up https://github.com/sul-dlss/argo/tree/profiling_and_indexing_improvements last night, i ran across my note to self about consolidating the dor_controller.rb code and the indexer.rb code.  and i think i got that working, though i need to fix and add some unit tests.  i'll push my WIP, since the commits need more grooming anyway, and you can take a look and tell me what you think... i assumed synchronous commit in my code, but it'd be easy enough to make that optional...